Fix evaluate

amj opened the pull request:
Rewrites the evaluate command to actually work w/ the new trainer rewrite and the new vlosses rewrite.

At some point, the 'name' field was removed from the DualNetwork object, although not actually removed from places that either set or referenced it.  Happily, those codepaths did not diverge...
Perhaps the player object should keep track of its semantic 'name', distinct from the filename/path used to reference its network files, and also distinct from the identifier it uses for GTP.

amj commented:
/retest

amj commented:
/retest

Kashomon commented on local_rl_loop.py:
```diff
@@ -79,6 +80,10 @@ def rl_loop():
             readouts=10)
 
+        print("See sgf files here?")
+        sgf_listing = subprocess.check_output("ls -l {}/full".format(sgf_dir).split())
```
Just use https://docs.python.org/3/library/os.html#os.listdir and filter?

amj replied:
see below; i'd like to see these are nonzero size files.

brilee replied:
https://docs.python.org/3/library/os.html#os.scandir  is the equivalent of `ls -l` in pure python

brilee replied:
also, `subprocess.X(some_string.split())` is an antipattern. If `some_string` contains an unintended space, it would be the equivalent of passing in a filename to a bash command where the filename had a space in it. In bash, you would quote the filename; the way python would know to quote the filename is by passing in a list as you intended it to be split: `subprocess.check_output(['ls', '-l', filename])`

Kashomon commented on local_rl_loop.py:
```diff
@@ -79,6 +80,10 @@ def rl_loop():
             holdout_pct=100,
             readouts=10)
 
+        print("See sgf files here?")
```
Not sure I understand this comment.

amj replied:
these are 'integration tests' with big scare quotes.  If you're running this and you want to see that everything still works, you should see some files with nonzero size.

amj commented on evaluation.py:
```diff
@@ -14,84 +14,80 @@
             move = active.pick_move()
             active.play_move(move)
             inactive.play_move(move)
+            print(active.root.position)
```
Good point, i should check the verbosity level.

brilee commented on evaluation.py:
```diff
@@ -14,84 +14,82 @@
-            inactive = black if num_moves % 2 else white
+        while True:
+            start = time.time()
+            for _ in range(readouts):
```
this should read `while player.num_readouts < readouts`, because parallel vloss search will add up to 8 readouts per tree_search call.

amj replied:
Added.

brilee commented on evaluation.py:
```diff
@@ -14,84 +14,82 @@
+        while True:
+            start = time.time()
+            for _ in range(readouts):
+                player = white if num_move % 2 else black
```
this should happen outside of for loop,

amj replied:
Oops, right.

brilee commented on evaluation.py:
```diff
@@ -14,84 +14,82 @@
+            if verbosity >= 3:
+                print(player.root.position)
+
+            active = white if num_move % 2 else black
```
this computation should happen once at top

amj replied:
Done

brilee commented on evaluation.py:
```diff
@@ -14,84 +14,82 @@
+                fname = "{:d}-{:s}-vs-{:s}-{:d}.sgf".format(int(time.time()),
+                                                            white_name, black_name, i)
+                with open(os.path.join(sgf_dir, fname), 'w') as _file:
+                    sgfstr = sgf_wrapper.make_sgf(active.position.recent,
```
I could have sworn I had some cute code snippet that would splice in alternate comments from both players so you'd get to see both players' debug perspectives. Let me figure out where I put it...

brilee commented on evaluation.py:
```diff
@@ -14,84 +14,82 @@
-                continue
+                active.set_result(-1 *
+                                  active.root.position.to_play, was_resign=True)
+                inactive.set_result(
```
do you need to set this result if you don't use inactive's result string?

brilee commented on evaluation.py:
```diff
@@ -14,84 +14,82 @@
+                inactive.set_result(
+                    active.root.position.to_play, was_resign=True)
+
+            if active.is_done():
```
seems like you could just remove this check and fuse the contents of this IF with the contents of the IF above. Then you wouldn't need is_done() either.

amj commented:
'scandir' is not quite what i want.  i just want it to spit out 'ls -l' so i can verify there's some kind of nonzero-sized files.  I don't want to check what they are, or read them, or make any other assertions other than 'looks like small SGFs were written'.  

Re: subprocess.x(foo.split()) right right right, fixed.
